### PR TITLE
Release preparation

### PR DESCRIPTION
**Pull request by mkorszun**
* Remove support for elixir `1.6` and `1.7`
* Set minimum elixir version to `1.7`
* Reformat code
* Update `ex_doc`
* Bump `gen_rmq` version to `2.0.0`
* Disable `dialyzer` - travis builds are terminating while building PLTs. Need to find a better way for PLT generation / caching

## Checklist

- [ ] I have added unit tests to cover my changes.
- [ ] I have improved the code quality of this repo. (refactoring, or reduced number of static analyser issues)
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -1,13 +1,13 @@
 defmodule GenRMQ.Mixfile do
   use Mix.Project
 
-  @version "1.3.0"
+  @version "1.4.0"
```
Comment (vorce) [inline mix.exs]: Not 100% sure but I think in terms of SemVer we have to bump the major version

Comment (mkorszun) [reply mix.exs]: Then maybe we should consider bumping directly to 1.7 - what do you think? Is it too much to stop supporting two versions in one release :)?

Comment (vorce) [reply mix.exs]: Let's go to 1.7 directly. Makes more sense to me.

File: .travis.yml [hunk 2]
```diff
@@ -1,27 +1,21 @@
...
-
 sudo: required
 
 services:
   - docker
 
 before_script:
   - make code-analysis
-  - make dialyzer
+  # - make dialyzer
```
Comment (vorce) [inline .travis.yml]: Reason for disabling it?

File: mix.exs [hunk 3]
```diff
@@ -1,13 +1,13 @@
 defmodule GenRMQ.Mixfile do
   use Mix.Project
 
-  @version "1.3.0"
+  @version "2.0.0"
```
Comment (vorce) [inline mix.exs]: Do we also need to update the README?

File: .travis.yml (hunk 2 above)
Comment (mkorszun) [reply .travis.yml]: @vorce The builds are timeouting atm :/ Need to spend more time on it. Do not want it to block this release though.

File: mix.exs (hunk 3 above)
Comment (mkorszun) [reply mix.exs]: @vorce It is here https://github.com/meltwater/gen_rmq/pull/103/files?short_path=04c6e90#diff-04c6e90faac2675aa89e2176d2eec7d8. Are you thinking about smth more?

Comment (vorce) [reply mix.exs]: @mkorszun Ah ok, no that is what I meant 👍

**Pull request merged**